### PR TITLE
Add stop scan functionality

### PR DESCRIPTION
**Pull request by AllForNothing**
Related back-end PR #15548
Signed-off-by: AllForNothing <sshijun@vmware.com>

### CODE REVIEW

File: src/portal/src/app/base/left-side-nav/system-robot-accounts/system-robot-util.ts [hunk 1]
```diff
@@ -92,6 +92,11 @@ export const INITIAL_ACCESSES: FrontAccess[] = [
         "resource": "scan",
         "action": "create",
         "checked": true
+    },
+    {
+        "resource": "scan",
+        "action": "delete",
```
Comment (zyyw) [inline src/portal/src/app/base/left-side-nav/system-robot-accounts/system-robot-util.ts:98]: stop scan action has changed from “delete" to "stop" (a new action type being added), you may want to update it as well. 
please refer to "Robot account permission" the proposal for more details: 
`https://github.com/goharbor/community/blob/4cb9f33e578cc2cf59c6b89d0172efebefb33746/proposals/new/stop-scan-job-proposal.md#robot-account-permission`
`https://github.com/goharbor/community/blob/4cb9f33e578cc2cf59c6b89d0172efebefb33746/proposals/new/stop-scan-job-proposal.md#robot-account-permission-1`

File: src/portal/src/app/base/left-side-nav/interrogation-services/vulnerability/vulnerability-config.component.html [hunk 2]
```diff
@@ -13,10 +13,10 @@
     <div class="clr-row">
     <div class="clr-col-2 flex-200">
         <div class="btn-scan-right btn-scan margin-top-16px">
-            <button id="scan-now" class="btn btn-primary  btn-scan" (click)="scanNow()"
-                    [disabled]="!scanAvailable">
-                <span *ngIf="scanAvailable">{{ 'CONFIG.SCANNING.SCAN_NOW' | translate }}</span>
-                <span *ngIf="!scanAvailable">{{ 'CONFIG.SCANNING.SCAN' | translate }}</span>
+            <button id="scan-now" class="btn btn-primary  btn-scan" (click)="scanOrStop()"
```
Comment (steven-zou) [inline src/portal/src/app/base/left-side-nav/interrogation-services/vulnerability/vulnerability-config.component.html:16]:
```suggestion
            <button id="scan" class="btn btn-primary  btn-scan" (click)="scanOrStop()"
```

File: src/portal/src/app/base/left-side-nav/interrogation-services/vulnerability/vulnerability-config.component.spec.ts [hunk 3]
```diff
@@ -104,5 +106,11 @@ describe('VulnerabilityConfigComponent', () => {
      component.scanningMetrics.ongoing = false;
      component.hasDefaultScanner = true;
      expect(component.scanAvailable).toBeTruthy();
+     const spnManual: HTMLSpanElement = fixture.nativeElement.querySelector(".float-left");
+     expect(spnManual.innerText).toEqual('CONFIG.SCANNING.SCHEDULED');
+     const scanBtn: HTMLButtonElement = fixture.nativeElement.querySelector("#scan-now");
```
Comment (steven-zou) [inline src/portal/src/app/base/left-side-nav/interrogation-services/vulnerability/vulnerability-config.component.spec.ts:111]:
```suggestion
     const scanBtn: HTMLButtonElement = fixture.nativeElement.querySelector("#scan");
```

File: src/portal/src/app/base/project/repository/artifact/artifact-additions/artifact-vulnerabilities/artifact-vulnerabilities.component.html [hunk 4]
```diff
@@ -11,14 +11,16 @@
             <clr-dg-action-bar>
                 <div class="clr-row center">
                     <div class="ml-05">
-                        <button id="scan-btn" (click)="scanNow()" type="button"  class="btn  btn-secondary" [clrLoading]="scanBtnState" [disabled]="!(hasEnabledScanner && hasScanningPermission && !onSendingScanCommand)">
-                            <clr-icon shape="shield-check" size="16"></clr-icon>&nbsp;
-                            <span>{{'VULNERABILITY.SCAN_NOW' | translate}}</span>
+                        <button id="scan-btn" (click)="scanOrStop()" type="button"  class="btn  btn-secondary" [clrLoading]="scanBtnState" [disabled]="!(hasEnabledScanner && hasScanningPermission && !onSendingScanCommand)">
```
Comment (steven-zou) [inline src/portal/src/app/base/project/repository/artifact/artifact-additions/artifact-vulnerabilities/artifact-vulnerabilities.component.html:14]: a method can be defined to represent the button status conditions: `!(hasEnabledScanner && hasScanningPermission && !onSendingScanCommand)`

**Review by zyyw — CHANGES_REQUESTED** [review]

File: src/portal/src/app/base/project/repository/artifact/artifact-list-page/artifact-list/artifact-list-tab/artifact-list-tab.component.ts [hunk 5]
```diff
@@ -1108,7 +1144,7 @@ export class ArtifactListTabComponent implements OnInit, OnDestroy {
               if (res.headers) {
                 let xHeader: string = res.headers.get("x-total-count");
                 if (xHeader) {
-                  item.tagNumber = Number.parseInt(xHeader);
+                  item.tagNumber = Number.parseInt(xHeader, 10);
```
Comment (steven-zou) [inline src/portal/src/app/base/project/repository/artifact/artifact-list-page/artifact-list/artifact-list-tab/artifact-list-tab.component.ts:1147]: If the `xHeader` is not a number?

**Review by steven-zou — COMMENTED** [review]
comments left

File: src/portal/src/app/base/left-side-nav/system-robot-accounts/system-robot-util.ts (hunk 1 above)
Comment (AllForNothing) [reply src/portal/src/app/base/left-side-nav/system-robot-accounts/system-robot-util.ts]: Updated

File: src/portal/src/app/base/project/repository/artifact/artifact-additions/artifact-vulnerabilities/artifact-vulnerabilities.component.html (hunk 4 above)
Comment (AllForNothing) [reply src/portal/src/app/base/project/repository/artifact/artifact-additions/artifact-vulnerabilities/artifact-vulnerabilities.component.html:14]: updated

File: src/portal/src/app/base/left-side-nav/interrogation-services/vulnerability/vulnerability-config.component.spec.ts (hunk 3 above)
Comment (AllForNothing) [reply src/portal/src/app/base/left-side-nav/interrogation-services/vulnerability/vulnerability-config.component.spec.ts]: Updated

File: src/portal/src/app/base/project/repository/artifact/artifact-list-page/artifact-list/artifact-list-tab/artifact-list-tab.component.ts (hunk 5 above)
Comment (AllForNothing) [reply src/portal/src/app/base/project/repository/artifact/artifact-list-page/artifact-list/artifact-list-tab/artifact-list-tab.component.ts:1147]: I think the back-end API  should handle this.  if x-total-count is set, it should be a number bigger than zero

File: src/portal/src/app/base/left-side-nav/interrogation-services/vulnerability/vulnerability-config.component.html (hunk 2 above)
Comment (AllForNothing) [reply src/portal/src/app/base/left-side-nav/interrogation-services/vulnerability/vulnerability-config.component.html]: Updated

**Review by zyyw — APPROVED** [review]
lgtm

**Review by steven-zou — APPROVED** [review]
lgtm

**Pull request merged**